python/lsst/meas/algorithms/setPrimaryFlags.py

```diff
@@ -295,15 +295,20 @@ def getDeblendPrimaryFlags(sources):
         # the number of deblended sources in the catalog,
         # because some peaks might have been culled during deblending.
         nPeaks = sources["deblend_nPeaks"]
-        parentNChild = sources["deblend_parentNChild"]
+        nChild = sources["deblend_nChild"]
+        parentNPeaks = sources["deblend_parentNPeaks"]
+        depth = sources["deblend_depth"]
         # It is possible for a catalog to contain a hierarchy of sources,
         # so we mark the leaves (end nodes of the hierarchy tree with no
         # children).
-        isLeaf = nPeaks == 1
-        fromBlend = parentNChild > 1
-        isIsolated = isLeaf & ((parent == 0) | parentNChild == 1)
-        isDeblendedSource = (fromBlend & isLeaf) | (isIsolated & (parent == 0))
-        isDeblendedModelSource = (parent != 0) & isLeaf
+        isLeaf = nChild == 0
+        isChild = depth > 0
+        isSibling = parentNPeaks > 1
+        isDeblendedModelSource = isLeaf & isChild
+        fromBlend = isDeblendedModelSource & isSibling
+        isIsolatedParent = (depth == 0) & (nPeaks == 1)
+        isIsolated = isIsolatedParent | ((depth == 1) & ~isSibling)
```

Comment:
Shouldn't this generally be `depth >= 1`?

Reply:
No, anything with depth > 1 came from a blend, so by definition it can't be a model of an isolated source. It would be a source that is isolated in the deconvolved frame but not the observed seeing.